src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -64,69 +64,92 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         co_await winrt::resume_foreground(Dispatcher());
 
-        // "remove" all profile-related NavViewItems
-        // LOAD-BEARING: use Visibility here, instead of menuItems.Remove().
-        //               Remove() works fine on NavViewItems with an hstring tag,
-        //               but causes an out-of-bounds error with Profile tagged items.
-        //               The cause of this error is unknown.
+        // Deduce information about the currently selected item
+        IInspectable selectedItemTag;
         auto menuItems{ SettingsNav().MenuItems() };
-        for (auto i = menuItems.Size() - 1; i > 0; --i)
+        if (const auto& selectedItem{ SettingsNav().SelectedItem() })
         {
-            if (const auto navViewItem{ menuItems.GetAt(i).try_as<MUX::Controls::NavigationViewItem>() })
+            if (const auto& navViewItem{ selectedItem.try_as<MUX::Controls::NavigationViewItem>() })
             {
-                if (const auto tag{ navViewItem.Tag() })
+                selectedItemTag = navViewItem.Tag();
+            }
+        }
+
+        // remove all profile-related NavViewItems by populating a std::vector
+        // with the ones we want to keep.
+        // NOTE: menuItems.Remove() causes an out-of-bounds crash. Using ReplaceAll()

```

Comment:
OF COURSE IT DOES.

Is there an upstream bug for the `Remove` crash?

Reply:
No. I couldn't get a repro in a custom XAML app last month. The crash doesn't happen when you removed hstring tagged items, only `Profile` tagged items. Not exactly sure how to report that. Maybe it's XAML Islands related? 🤷‍♂️

Reply:
GAH. Okay. Well, eff it then.